pkg/crc/preflight/preflight_checks_windows.go

```diff
@@ -99,40 +100,27 @@ func checkHyperVServiceRunning() error {
 	return nil
 }
 
-func checkIfUserPartOfHyperVAdmins() error {
-	// https://support.microsoft.com/en-us/help/243330/well-known-security-identifiers-in-windows-operating-systems
-	// BUILTIN\Hyper-V Administrators => S-1-5-32-578
-
-	checkIfMemberOfHyperVAdmins :=
-		`$sid = New-Object System.Security.Principal.SecurityIdentifier("S-1-5-32-578")
-	@([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole($sid)`
-	stdOut, _, err := powershell.Execute(checkIfMemberOfHyperVAdmins)
+func checkUserPartOfCrcUsersAndHypervAdminsGroup() error {
+	var mErr errors.MultiError
+	_, _, err := powershell.Execute(fmt.Sprintf("Get-LocalGroupMember -Group 'crc-users' -Member '%s'", username()))
 	if err != nil {
-		logging.Debug(err.Error())
-		return fmt.Errorf("Failed checking if user is part of hyperv admins group")
-	}
-	if !strings.Contains(stdOut, "True") {
-		return fmt.Errorf("User is not a member of the Hyper-V administrators group")
+		logging.Debugf("current user is not part of 'crc-users' group: %v", err)
+		mErr.Collect(err)
 	}
-
-	return nil
-}
-
-func fixUserPartOfHyperVAdmins() error {
-	_, _, err := powershell.ExecuteAsAdmin("adding current user to Hyper-V administrator group", fmt.Sprintf("Add-LocalGroupMember -SID 'S-1-5-32-578' -Member '%s'", username()))
+	_, _, err = powershell.Execute(fmt.Sprintf("Get-LocalGroupMember -SID 'S-1-5-32-578' -Member '%s'", username()))
 	if err != nil {
-		return err
+		logging.Debugf("current user is not part of 'Hyper-V Administrator' group: %v", err)
+		mErr.Collect(err)
 	}
-	return errReboot
-}
-
-func checkUserPartOfCrcUsers() error {
-	_, _, err := powershell.Execute(fmt.Sprintf("Get-LocalGroupMember -Group 'crc-users' -Member '%s'", username()))
-	return err
+	if len(mErr.Errors) == 0 {
+		return nil
+	}
+	return fmt.Errorf("Current user is not part of crc-users or hyper-v admins group: %s", mErr.Error())
 }
 
-func fixUserPartOfCrcUsers() error {
-	_, _, err := powershell.ExecuteAsAdmin("adding current user to crc-users group", fmt.Sprintf("Add-LocalGroupMember -Group 'crc-users' -Member '%s'", username()))
+func fixUserPartOfCrcUsersAndHypervAdminsGroup() error {
+	cmd := `Add-LocalGroupMember -Group 'crc-users' -Member '%s';Add-LocalGroupMember -SID 'S-1-5-32-578' -Member '%s'`
```

Comment:
If the user is already part of `crc-users`, but not of  `S-1-5-32-578` does this work as expected? I mean, is `Add-LocalGroupMember -Group 'crc-users'` a no-op if we are already part of that group?

Reply:
yes, its not a no-op it'll provide an error but since we are not checking the error reported by powershell, and powershell's default behaviour when an error occurs is to report and continue (`
$ErrorActionPreference=Continue`) it goes on to execute the next command